client/src/stores/collectionElementsStore.ts

```diff
@@ -1,13 +1,36 @@
 import { defineStore } from "pinia";
-import Vue, { computed, ref } from "vue";
+import { computed, del, ref, set } from "vue";
 
 import { CollectionEntry, DCESummary, HDCASummary, HistoryContentItemBase, isHDCA } from "./services";
-import * as Service from "./services/datasetCollection.service";
+import { fetchCollectionDetails, fetchElementsFromCollection } from "./services/datasetCollection.service";
+
+/**
+ * Represents an element in a collection that has not been fetched yet.
+ */
+export interface ContentPlaceholder {
+    /**
+     * The index of the element in the containing collection.
+     *
+     * This is used to determine the offset to fetch the element from when
+     * scrolling through the collection.
+     */
+    element_index?: number;
```

Comment:
Very minor: Don't you always know the element_index ? Similarly, could you require fetching to be a boolean and init it with true ?

Reply:
Good point, making `element_index` optional here was just to be able to make it work with `Array.fill` but now that I look at it we still need to map the index to each placeholder so making it required and using simple for loop would be even better. I'll go and do that :+1: 

About making the `fetching` boolean default to true... I'm not so sure, the idea here was to know exactly which placeholder was currently being fetched instead of just sitting there at the middle or end of the collection, which may never be fetched. In my head, I was accounting for possible multiple fetches in parallel (which I don't think will happen in JS though?) so we can skip those placeholders that are already waiting for a real value. Also just to save a tiny bit of memory if you have a huge number of elements :sweat_smile: